src/base/ApplicationSQSQueue.ts

```diff
@@ -0,0 +1,144 @@
+import { Resource } from 'cdktf';
+import { Construct } from 'constructs';
+import { SqsQueue } from '../../.gen/providers/aws';
+
+export interface ApplicationSQSQueueProps {
+  /**
+   * Name of the sqs queue
+   */
+  name?: string;
+  /**
+   * The number of seconds Amazon SQS retains a message.
+   */
+  messageRetentionSeconds?: number;
+  /**
+   * How many times a message can be received before it goes to a dead letter queue
+   * If not set or set to 0 this will instead max messages constantly retry
+   */
+  maxReceiveCount?: number;
+  /**
+   * Max size the SQS message can be. In Bytes
+   */
+  maxMessageSize?: number;
+  /**
+   * How long should a message wait before its allowed to be visible to a worker
+   */
+  delaySeconds?: number;
+  /**
+   * Once a message is being worked on, how long can the worker hold it.
+   */
+  visibilityTimeoutSeconds?: number;
+  /**
+   * How long should a worker be allowed to long poll
+   */
+  receiveWaitTimeSeconds?: number;
+  tags?: { [key: string]: string };
+}
+
+/**
+ * Creates an SQS Queue with a dead letter queue
+ */
+export class ApplicationSQSQueue extends Resource {
+  public readonly sqsQueue: SqsQueue;
+
+  constructor(
+    scope: Construct,
+    name: string,
+    config: ApplicationSQSQueueProps
+  ) {
+    super(scope, name);
+    ApplicationSQSQueue.validateConfig(config);
+    this.sqsQueue = this.createSQSQueue(config);
+  }
+
+  /**
+   * Validates the config against the values defined in:
+   * https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sqs_queue
+   * @param config
+   * @private
+   */
+  private static validateConfig(config: ApplicationSQSQueueProps): void {
+    if (
+      config.visibilityTimeoutSeconds &&
+      (config.visibilityTimeoutSeconds < 0 ||
+        config.visibilityTimeoutSeconds > 43200)
+    ) {
+      throw new Error(
+        'Visibility timeout can not be greater then 43200 or less then 0'
+      );
+    }
+
+    if (
+      config.messageRetentionSeconds &&
+      (config.messageRetentionSeconds > 43200 ||
+        config.messageRetentionSeconds < 60)
+    ) {
+      throw new Error(
+        'Message retention can not be greater then 1209600 or less then 60'
+      );
+    }
+
+    if (
+      config.maxMessageSize &&
+      (config.maxMessageSize > 262144 || config.maxMessageSize < 1024)
```

Comment:
is there a better way to handle these magic numbers? (there might not be.) can we add links to docs or something? and/or comments explaining the numbers here?

Reply:
@jpetto i put a link to the docs in the function comment does that suffice? (I also wish terraform cdk validated this for us... perhaps we pr to them..)